src/tools/pub/custom-package-repositories.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Custom Package Repositories
+description: "How Dart's package management tool, pub, works with custom package repositories."
+---
+
+The `pub` tool has support for third-party package repositories. A package
+repository is an HTTP server from which the `pub` client can fetch packages.
+The official package repository is [pub.dev](https://pub.dev), this repository
+is by the `pub` tool by default. A package repository is identified by a
+_hosted-url_ (e.g. `https://pub.example.com/my-account/`). 
+
+Third-party package servers can be useful for hosting private packages. It is
+also common to use [git-dependencies](./dependencies#git-packages) for hosting
+private packages, however, the `pub` does not support resolving versions against
+a git-repository it just fetches a specific revision of the git-repository.
+Hence, when many people are collaborating it is sometimes preferable to use a
+private package repository.
+
+
+## Authenticating with a custom package repository
+Custom package servers are often private and require authentication. The `pub`
+tool authenticates against package repositories by attaching a secret token to
+the requests. When the `pub` tool does not have a token for a given
+_hosted-url_ it will attempt to make requests without authentication.
+
+To specify a secret token for the `pub` tool to use with a given _hosted-url_,
+use the `dart pub token add <hosted-url>` command.
+
+
+## Getting dependencies from a custom package repository
+To fetch a package from custom package repository we must specify the
+_hosted-url_ for the package in `pubspec.yaml`, using the syntax for
+[hosted packages](./dependencies#hosted-packages).
+
+{% prettify yaml tag=pre+code %}
+dependencies:
+  foo:
+    hosted: https://pub.example.com
+    version: ^1.4.0
+{% endprettify %}
```

Comment:
````suggestion
```yaml
dependencies:
  foo:
    hosted: https://pub.example.com
    version: ^1.4.0
```
````

Reply:
I didn't know we could do that, we should probably patch the other pages -- I just cargo culted.. :D